src/datasources/cache/redis.cache.service.ts

```diff
@@ -31,7 +31,12 @@ export class RedisCacheService
     return this.client.ping();
   }
 
-  async set(
+  async getCounter(key: string): Promise<number | null> {
+    const value = await this.client.get(this._prefixKey(key));
+    return value ? Number(value) || null : null;
```

Comment:
If the cache returns `'0'`, this will return `null`.

Reply:
True 🙈

Fixed and tested in [49b696e](https://github.com/safe-global/safe-client-gateway/pull/1948/commits/49b696e7f63c8d1dea67472f39b3136c122a1ce0)